src/app/errorReports/newErrorReportForm/NewErrorReportForm.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { postErrorReport, getLocation } from '../../../actions';
+import { selectIsPostingErrorReport } from '../../../reducers';
+import { selectLocationError, selectLocationData } from '../../../selectors/location';
+import LoadingLabel from '../../../components/form/LoadingLabel';
+import ErrorLabel from '../../../components/form/ErrorLabel';
+import ErrorReportText from './ErrorReportText';
+import ErrorReportServices from './ErrorReportServices';
+
+class NewErrorReportForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errorReportText: '',
+      errorReportGeneralLocationError: false,
+      errorReportServices: [],
+      isErrorReportFinished: false,
+    };
+
+    this.onErrorReportTextChanged = this.onErrorReportTextChanged.bind(this);
+    this.onErrorReportServicesChanged = this.onErrorReportServicesChanged.bind(this);
+    this.onErrorReportGeneralLocationErrorChanged = this.onErrorReportGeneralLocationErrorChanged.bind(this);
+    this.onErrorReportFinished = this.onErrorReportFinished.bind(this);
+    this.onErrorReportSubmitted = this.onErrorReportSubmitted.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.props.locationData) {
+      this.props.getLocation(this.props.match.params.locationId);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.isPostingErrorReport && !this.props.isPostingErrorReport) {
+      this.props.history.push(`/errorreports/${this.props.match.params.locationId}/thanks`);
+    }
+  }
+
+  onErrorReportTextChanged(text) {
+    this.setState({ errorReportText: text });
+  }
+
+  onErrorReportServicesChanged(value) {
+    this.setState((state) => {
+      const errorReportServices = state.errorReportServices.includes(value)
+        ? state.errorReportServices.filter(service => service !== value)
+        : [...state.errorReportServices, value];
+      return {
+        errorReportServices,
+      };
+    });
+  }
+
+  onErrorReportGeneralLocationErrorChanged(value) {
+    this.setState({ errorReportGeneralLocationError: value });
+  }
+
+  onErrorReportFinished() {
+    this.setState({ isErrorReportFinished: true });
```

Comment:
This naming is confusing, since it's not the error report which is finished, but only the step where the user indicates where the errors are.
The state and function should be renamed to reflect this.

Reply:
Yeah, I thought the same, and considered renaming it, however I was trying to remain consistent with the `comments` naming.

Perhaps something like `isErrorReportFinalScreen` would be more verbose.

Reply:
Sure, that sounds good.
Following that thinking, you could be even more explicit, with something like "isOnErrorContentScreen".
Anything that suggests only the selection of services is done but the report isn't quite done yet.